lib/driver.js

```diff
@@ -322,6 +322,18 @@ class AndroidDriver extends BaseDriver {
     Object.assign(this.opts, launchInfo);
     Object.assign(this.caps, launchInfo);
 
+    // Uninstall any uninstallOtherPackages which were specified in caps
+    if (this.opts.uninstallOtherPackages) {
+      let uninstallOtherPackages;
+      try {
+        uninstallOtherPackages = helpers.parseArray(this.opts.uninstallOtherPackages);
```

Comment:
I'd prefer this stuff be in the [validateDesiredCaps](https://github.com/appium/appium-android-driver/blob/master/lib/android-helpers.js#L702-L718) function.